src/routes/faucet.ts

```diff
@@ -39,10 +40,44 @@ export type OverlayTx = {
   nextBlockHashBee: string
 }
 
+interface OldGasOptions {
+  gasPrice: BigNumberish
+}
+interface EIP1559GasOptions {
+  maxFeePerGas: BigNumberish
+  maxPriorityFeePerGas?: BigNumberish
+}
+
+type TxGasOptions = OldGasOptions | EIP1559GasOptions
+
 // Errors
 export class HasTransactionsError extends Error {}
 export class BlockTooRecent extends Error {}
 
+// Constants
+const gasMult = 2 // This must be an integer value because we don't do any fancy rounding and it would throw runtime errors
+
+async function getGasPrice(wallet: Wallet): Promise<TxGasOptions> {
+  const gasPrice = await wallet.getGasPrice()
+  const feeData = await wallet.getFeeData()
+  let txGasOptions: TxGasOptions
+
+  // The blockchain supports EIP-1559
+  if (feeData.maxFeePerGas !== null && feeData.maxPriorityFeePerGas !== null) {
+    txGasOptions = {
+      maxFeePerGas: feeData.maxFeePerGas,
+      maxPriorityFeePerGas: feeData.maxPriorityFeePerGas.mul(gasMult),
+    }
+  }
+
+  // Default to gasPrice
+  else {
+    txGasOptions = { gasPrice: gasPrice.mul(gasMult) }
+  }
+
+  return txGasOptions
+}
+
```

Comment:
So this implementation is quite wasteful, it doubles the transaction price for blockchains that do not support EIP-1559 and it doubles the priority fee on blockchains that do support it. However, with @agazso we agreed that the TX costs are not significant and hence having them by default double is OK.